### DISCUSSION TITLE
Excluding a single page from nav

### DISCUSSION THREAD

**Discussion by madjxatw** [squidfunk/mkdocs-material#6863] [Q&A]
I'm currently using literate-nav plugin to include directories to nav instead of listing files one by one, however, it doesn't support my special need that excludes a specific file from nav but still keeps it in the site. Is there any plugin that can meet my requirement?

**Comment by alexvoss** [comment]
Have you tried the [awesome pages plugin](https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/)? It might have the kind of filtering functions you need - at least if you are at liberty to rename files.

**Reply by kamilkrzyskow** [reply]
They actually made a post there without mentioning: 
- https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/issues/102

~~There is the new `excluded_docs` config option: https://www.mkdocs.org/user-guide/configuration/#exclude_docs~~
But it probably works only with the default `mkdocs.yml` `nav` logic, not with the awesome-pages-plugin.

Actually it completely excludes the file, not just in the navigation, so that's probably not the solution here.

**Reply by madjxatw** [reply]
I've checked the awesome-pages plugin but did not see any example covering my requirement. A question has been created there and I'am waiting for an answer.

@kamilkrzyskow , I've also checked mkdocs user guide and even thought that `not_in_nav` is the way to go. Unfortunately, it is used just for suppressing the warning about files not included in nav :-(

**Reply by alexvoss** [reply]
Are you sure the [Rest Filter Patterns](https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/?tab=readme-ov-file#rest-filter-patterns) would not do what you need? If you can freely name the files then it should be possible to make the file that is to be excluded not match a pattern while everything else does.

**Reply by madjxatw** [reply]
Then I have to rename all the files to be matched in a certain pattern, which is not what I expect. I prefer a way like how `.gitignore` does so that I could simply prefix the excluded file with a `!` or something else.

**Reply by alexvoss** [reply]
> Then I have to rename all the files to be matched in a certain pattern

That pattern could be very generic, though. The point would be to *not* match the one file you want to exclude. A regex pattern should be able to do that.

I can see how you might want to have an exclude pattern. For us this is an upstream (MkDocs) or plugin issue, not something we have a way to address, really.

**Reply by kamilkrzyskow** [reply]
> I've also checked mkdocs user guide and even thought that not_in_nav is the way to go. Unfortunately, it is used just for suppressing the warning about files not included in nav :-(

https://github.com/mkdocs/mkdocs/blob/e755aaed7ea47348a60495ab364d5483ab90a4a6/mkdocs/structure/nav.py#L134-L136
According to this, if the `nav` is not set in the mkdocs.yml file then the `not_in_nav` setting should influence the gathering of the nav items.
It won't exclude them from nav if it's defined already. 🤔Sorry for asking, but have you tried it? Of course without the awesome_pages plugin.

As for the awesome-pages plugin, from the docs it's hard to say if it supports groups in regex patterns, so that you could in a way write name **inclusion** lists as a regex patterns `(name\.md)|(some-other-name\.md)`, as long there aren't too many files per directory then the performance shouldn't degrade too much. So every file not matched would not be included. However, I can see that defeats the purpose of automatic nav generation 😞
